src/lib/states/task-ecs.js

```diff
@@ -0,0 +1,120 @@
+const AWS = require('aws-sdk');
+const Task = require('./task');
+const { pascalCaseToCamelCase } = require('../tools/case');
+const addHistoryEvent = require('../actions/custom/add-history-event');
+
+class TaskEcs extends Task {
+  async invokeEcsTask() {
+    const ecsConfig = {};
+    if (this.config && this.config.ecsEndpoint) {
+      ecsConfig.endpoint = this.config.ecsEndpoint;
+    }
+    if (this.config && this.config.ecsRegion) {
+      ecsConfig.region = this.config.ecsRegion;
+    }
+    const ecs = new AWS.ECS(ecsConfig);
+
+    this.validateParams();
+    const parameters = this.state.Parameters;
+
+    // Pass in only the supported parameters from this page:
+    // https://docs.aws.amazon.com/step-functions/latest/dg/connectors-ecs.html
+    // Note that we need to call pascalCaseToCamelCase on any objects below as Step
+    // Functions requires all object keys to be PascalCase, whereas the ECS API requires
+    // them all to be camelCase.
+    const params = {
+      cluster: parameters.Cluster,
+      group: parameters.Group,
+      launchType: parameters.LaunchType,
+      networkConfiguration: pascalCaseToCamelCase(parameters.NetworkConfiguration),
+      overrides: pascalCaseToCamelCase(parameters.Overrides),
+      placementConstraints: pascalCaseToCamelCase(parameters.PlacementConstraints),
+      placementStrategy: pascalCaseToCamelCase(parameters.PlacementStrategy),
+      platformVersion: parameters.PlatformVersion,
+      taskDefinition: parameters.TaskDefinition,
+    };
+
+    const runTaskResult = await ecs.runTask(params).promise();
+
+    if (runTaskResult.failures) {
+      throw new Error(`There were failures running the ECS Task for state ${this.name}: ${JSON.stringify(runTaskResult.failures)}`);
+    }
+
+    addHistoryEvent(this.execution, 'TASK_SCHEDULED');
+
+    if (this.useSync) {
+      const task = TaskEcs.getTaskFromEcsTaskResult(runTaskResult);
+      await this.waitForEcsTaskToFinish(ecs, task.taskArn);
+    }
+  }
+
+  async waitForEcsTaskToFinish(ecs, parameters, taskArn) {
+    // TODO: I've seen a history event called TaskSubmitted...
```

Comment:
Yea, I ran it and that's where I got the list of events I have. I did see `TaskSubmitted`, but there was no info on _when_ that's supposed to fire... It's _probably_ safer to omit it and leave the TODO until we have docs on these events rather than fire it at the wrong time, but I'm open to other suggestions.